src/ansys/mapdl/core/launcher.py

```diff
@@ -2004,3 +2048,153 @@ def _parse_ip_route(output):
+    # Sharing tasks acrros multiple nodes (DMP)
+    # the format of this envvar is a bit tricky. Avoiding it for the moment.
+    # SLURM_TASKS_PER_NODE = int(
+    #     kwargs.pop(
+    #         "SLURM_TASKS_PER_NODE", os.environ.get("SLURM_TASKS_PER_NODE", 1)
+    #     )
+    # )
+
+    # cpus-per-task is for multithreading,
+    # sharing tasks across multiple CPUs in same node (SMP)
+    SLURM_CPUS_PER_TASK = int(
+        kwargs.pop("SLURM_CPUS_PER_TASK", os.environ.get("SLURM_CPUS_PER_TASK", 1))
+    )
+    LOG.info(f"SLURM_CPUS_PER_TASK: {SLURM_CPUS_PER_TASK}")
+
+    # Set to value of the --ntasks option, if specified. See SLURM_NTASKS. Included for backwards compatibility.
+    SLURM_NPROCS = int(kwargs.pop("SLURM_NPROCS", os.environ.get("SLURM_NPROCS", 1)))
+    LOG.info(f"SLURM_NPROCS: {SLURM_NPROCS}")
+
+    # Number of CPUs allocated to the batch step.
+    SLURM_CPUS_ON_NODE = int(
+        kwargs.pop("SLURM_CPUS_ON_NODE", os.environ.get("SLURM_CPUS_ON_NODE", 1))
+    )
+    LOG.info(f"SLURM_CPUS_ON_NODE: {SLURM_CPUS_ON_NODE}")
+
+    SLURM_MEM_PER_NODE = kwargs.pop(
+        "SLURM_MEM_PER_NODE", os.environ.get("SLURM_MEM_PER_NODE", None)
+    )
+    LOG.info(f"SLURM_MEM_PER_NODE: {SLURM_MEM_PER_NODE}")
+
+    SLURM_NODELIST = kwargs.pop(
+        "SLURM_NODELIST", os.environ.get("SLURM_NODELIST", "")
+    ).lower()
+    LOG.info(f"SLURM_NODELIST: {SLURM_NODELIST}")
+
+    if not exec_file:
+        exec_file = os.environ.get("PYMAPDL_MAPDL_EXEC", None)
+
+    if not exec_file:
+        # We should probably make a way to find it.
+        # We will use the module thing
+        pass
+    LOG.info(f"Using MAPDL executable in: {exec_file}")
+
+    if not jobname:
+        jobname = os.environ.get("SLURM_JOB_NAME", "file")
+    LOG.info(f"Using jobname: {jobname}")
+
+    # Checking specific env var
+    if not nproc:
+        nproc = os.environ.get("PYMAPDL_NPROC", None)
+
+    if not nproc:
+        ## Attempt to calculate the appropriate number of cores:
+        # Reference: https://stackoverflow.com/a/51141287/6650211
+        # I'm assuming the env var makes sense.
+        #
+        # - SLURM_CPUS_ON_NODE is a property of the cluster, not of the job.
+        #
+        options = [
+            # 4,  # Fall back option
+            SLURM_CPUS_PER_TASK * SLURM_NTASKS,  # (CPUs)
+            SLURM_NPROCS,  # (CPUs)
+            # SLURM_NTASKS,  # (tasks) Not necessary the number of CPUs,
+            # SLURM_NNODES * SLURM_TASKS_PER_NODE * SLURM_CPUS_PER_TASK,  # (CPUs)
+            SLURM_CPUS_ON_NODE * SLURM_NNODES,  # (cpus)
+        ]
+        LOG.info(f"On SLURM number of processors options {options}")
+        nproc = max(options)
+
+    LOG.info(f"Setting number of CPUs to: {nproc}")
+
+    if not ram:
+        if SLURM_MEM_PER_NODE:
+            # RAM argument is in MB, so we need to convert
+            if SLURM_MEM_PER_NODE[-1] == "T":  # tera
+                ram = int(SLURM_MEM_PER_NODE[:-1]) * 10**6
```

Comment:
@germa89 . We are talking of decimal values, not bytes, correct? because otherwise 1 GB == 1024 MB.

Reply:
yep. You are right!